genienlp/data_utils/numericalizer.py

```diff
@@ -114,27 +117,32 @@ def _use_fast(self):
                (self._preprocess_special_tokens and self._pretrained_name in ALLOWED_FAST_TOKENIZERS_IF_PREPROCESSING))
     
 
-    def get_tokenizer(self, save_dir):
+    def get_tokenizer(self, save_dir, config, src_lang, tgt_lang):
+        tokenizer_args = {'do_lower_case': False, 'do_basic_tokenize': False, 'cache_dir': self._cache,
+                          'use_fast': self._use_fast(), 'src_lang': src_lang, 'tgt_lang': tgt_lang}
         if save_dir is not None:
-            config = AutoConfig.from_pretrained(self._pretrained_name)
-            self._tokenizer = AutoTokenizer.from_pretrained(save_dir,
-                                                            do_lower_case=False,
-                                                            do_basic_tokenize=False,
-                                                            config=config,
-                                                            cache_dir=self._cache,
-                                                            use_fast=self._use_fast())
+            tokenizer_args.update({'pretrained_model_name_or_path': save_dir, 'config': config})
         else:
-            self._tokenizer = AutoTokenizer.from_pretrained(self._pretrained_name,
-                                                            do_lower_case=False,
-                                                            do_basic_tokenize=False,
-                                                            cache_dir=self._cache,
-                                                            use_fast=self._use_fast())
-            
-        if isinstance(self._tokenizer, BertTokenizer):
+            tokenizer_args.update({'pretrained_model_name_or_path': self._pretrained_name})
```

Comment:
Do you not pass the config in this case?

Reply:
pretrained_name is enough to load the correct tokenizer.

Reply:
Then why do you need to pass the config from outside?